src/indexes.ts

```diff
@@ -522,19 +524,40 @@ class Index<T extends Record<string, any> = Record<string, any>> {
   }
 
   /**
-   * Delete multiples documents of an index
+   * Delete multiples documents of an index.
+   *
+   * @param params - Params value can be:
+   *
+   *   - DocumentsDeletionQuery: An object containing the parameters to customize
+   *       your document deletion. Only available in Meilisearch v1.2 and newer
+   *   - DocumentsIds: An array of document ids to delete
    *
-   * @param documentsIds - Array of Document Ids to delete
    * @returns Promise containing an EnqueuedTask
    */
   async deleteDocuments(
-    documentsIds: string[] | number[]
+    params: DocumentsDeletionQuery | DocumentsIds
   ): Promise<EnqueuedTask> {
-    const url = `indexes/${this.uid}/documents/delete-batch`
-
-    const task = await this.httpRequest.post(url, documentsIds)
+    // If params is of type DocumentsDeletionQuery
+    const isDocumentsDeletionQuery =
+      !Array.isArray(params) && typeof params === 'object'
+    const endpoint = isDocumentsDeletionQuery
+      ? 'documents/delete'
+      : 'documents/delete-batch'
+    const url = `indexes/${this.uid}/${endpoint}`
+
+    try {
+      const task = await this.httpRequest.post(url, params)
```

Comment:
Does the response of this `post` method vary?

Reply:
It is always a task, but if the request fails it goes to the catch

Reply:
One is a plain object one is an instance of the Task Class